### PR TITLE
:heavy_plus_sign: add abe_policy dependency

### PR DESCRIPTION
**Pull request by heavenboy8**
Use `abe_policy` instead of using `policy.rs`

I will update the `Cargo.toml`  when https://github.com/Cosmian/abe_policy will be published

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -26,6 +26,7 @@ python = ["interfaces", "pyo3"]
 wasm_bindgen = ["interfaces", "js-sys", "wasm-bindgen"]
 
 [dependencies]
+abe_policy = { path = "../abe_policy" }
```
Comment (ThibsG) [inline Cargo.toml:29]: It is a relative depedency, out of `abe_gpsw` project, so this cannot work.

You need to publish this dependency, Cargo prevents to publish any crate where any dependency is not public

**Review by ThibsG — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 1 above)
Comment (heavenboy8) [reply Cargo.toml:29]: I know that :D I'm waiting someone reviews `abe_policy` before pushing it. I will update the deps afterwards.

File: src/error.rs [hunk 2]
```diff
@@ -11,13 +11,15 @@ use cosmian_crypto_base::CryptoBaseError;
...
 
-#[derive(Error, Debug, Clone, PartialEq)]
+#[derive(Error, Debug)]
 pub enum FormatErr {
     //
     // External errors conversion
     #[error("{0}")]
     CryptoError(String),
     #[error(transparent)]
+    PolicyError(#[from] abe_policy::error::Error),
```
Comment (ThibsG) [inline src/error.rs:21]: I guess now we can remove some of the variants we have below, which are redundant with `abe_policy`'s error.

Also, let's expose `Error` at top level of `abe_policy`'s crate, so you can `use abe_policy::Error` without having to know how much deeper is the `Error` struct within `abe_policy`'s crate (if you move it internally, it's gonna be a breaking change for nothing basically)

Comment (heavenboy8) [reply src/error.rs]: Done

**Comment by ThibsG** [comment]
Thanks!

**Pull request merged**